tests/type_inference/test_for.py

```diff
@@ -0,0 +1,48 @@
+import astroid
+import nose
+from hypothesis import given, settings
+from typing import Callable
+import tests.custom_hypothesis_support as cs
+settings.load_profile("pyta")
+
+
+def test_for_list():
+    """Test whether visitors properly set the type constraint of the a For node representing for/else statement
+     iterating over a simple list.
+    """
+    program = f'for num in [1, 2, 3]:\n' \
+              f'    x = 3 + num\n' \
+              f'else:\n' \
```

Comment:
You don't need to include the `else` here (and in the example `nodes` documentation). That's not important here.